src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -134,15 +134,30 @@ public AzureContext SetCurrentContext(string tenantId)
             return _profile.Context;
         }
 
-        public AzureContext SetCurrentContext(string subscriptionId, string tenantId, bool verifySubscription = true)
+        public AzureContext SetCurrentContext(string subscriptionId, string subscriptionName, string tenantId)
         {
-            if (!string.IsNullOrWhiteSpace(tenantId))
+            AzureSubscription subscription;
+            string subscriptionFilter = string.IsNullOrWhiteSpace(subscriptionId) ? subscriptionName : subscriptionId;
+            if (!string.IsNullOrWhiteSpace(subscriptionId))
             {
-                _profile.SetContextWithCache(new AzureContext(
-                    _profile.Context.Subscription,
-                    _profile.Context.Account,
-                    _profile.Context.Environment,
-                    CreateTenant(tenantId)));
+                TryGetSubscriptionById(tenantId, subscriptionId, out subscription);
+            }
+            else
```

Comment:
Since this is  a public method, shouldn't we have a check here for subscriptionName not being null or whitespace?  And throw an ArgumentException if not?